src/overlays/actors/ovl_En_Test3/z_en_test3.h

```diff
@@ -7,11 +7,35 @@ struct EnTest3;
 
 typedef void (*EnTest3ActionFunc)(struct EnTest3*, PlayState*);
 
+#define KAFEI_GET_PARAM_1F(thisx) ((thisx)->params & 0x1F)
+#define KAFEI_GET_PARAM_1E0(thisx) (((thisx)->params >> 5) & 0xF)
+
+typedef struct {
+    /* 0x0 */ u8 unk_0;
+    /* 0x1 */ u8 unk_1;
+    /* 0x2 */ u16 textId;
+} EnTest3_struct_D78; // size = 0x4
+
 typedef struct EnTest3 {
-    /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x628];
-    /* 0x76C */ EnTest3ActionFunc actionFunc;
-    /* 0x770 */ char unk_770[0x648];
+    /* 0x000 */ Player player;
+    /* 0xD78 */ EnTest3_struct_D78* unk_D78;
+    /* 0xD7C */ Path* unk_D7C;
+    /* 0xD80 */ s32 unk_D80;
+    /* 0xD84 */ f32 unk_D84;
+    /* 0xD88 */ s8 unk_D88;
+    /* 0xD89 */ u8 unk_D89;
+    /* 0xD8A */ s16 unk_D8A;
+    /* 0xD8C */ u8 unk_D8C;
+    /* 0xD8D */ s8 unk_D8D;
+    /* 0xD8E */ s16 subCamId;
+    /* 0xD90 */ Player* unk_D90;
+    /* 0xD94 */ EnTest3ActionFunc unk_D94;
```

Comment:
Any reason why the actionFunc is not named?

Reply:
I'm not confident it actually is an action function. EnTest3 is full of callbacks of various kinds that I didn't figure out the purpose of. Perhaps even naming the type `ActionFunc` was too presumptuous.